### PR TITLE
[BUU] Unit Scale - basic implementation

### LINKED ISSUE #12005: [BUU] Change unit type

### PR DESCRIPTION
**Pull request by dacook**
### What? Why?

- First step of #12005  in order to progress

Unfortunately the original design was too challenging to implement, so in order to move forward we will continue with a simple solution, the same as the existing bulk product edit screen.

#### Screenshots
This is just some of the variations when interacting with the components
Screenshot 2024-03-06 at 1 52 32 pm Screenshot 2024-03-06 at 2 13 22 pm Screenshot 2024-03-06 at 1 52 19 pm Screenshot 2024-03-06 at 1 52 46 pm 
### What should we test?
* Selecting a weight or volume type and saving
* Selecting 'items' and entering an item name
   * item name is required

Switching between items and other will show/hide the text input.

### Release notes

Changelog Category (reviewers may add a label for the release notes):

- [ ] User facing changes
- [ ] API changes (V0, V1, DFC or Webhook)
- [ ] Technical changes only
- [x] Feature toggled

The title of the pull request will be included in the release notes.

### CODE REVIEW

**Review by rioug — APPROVED** [review]
Looks good ! I appreciate the preparatory work for the product refactor :pray:

File: app/services/weights_and_measures.rb [hunk 1]
```diff
@@ -45,9 +45,16 @@ def self.available_units
...
       measurement_info.filter do |_scale, unit_info|
         available_units.include?(unit_info['name'])
-      end.sort.to_h # sort by unit (hash key)
+      end.
+        # Remove duplicates by name
+        uniq do |_scale, unit_info|
+        unit_info['name']
+      end.
+        # Sort by unit number
+        sort.to_h
```
Comment (mkllnk) [inline app/services/weights_and_measures.rb:57]: If we remove duplicates by name then, I guess, it randomly selects one of the duplicate options. So we may get the low precision entry or the high precision entry. We haven't actually discussed which version we want to use.

**Review by mkllnk — APPROVED** [review]
Great!

The taxonomy fixes are random here but good.

**Comment by mariocarabotta** [comment]
thanks @filipefurtad0 this is fine for now. We'll tackle min width of all columns separately to fix those truncating / wrapping issues in one go!

cheers

**Pull request merged**